### ISSUE TITLE
Mailing queue stopped working due to invalid customer name

### ISSUE THREAD

**Issue by dandrikop** [magento/magento2#39432]
### Preconditions and environment

At Magento 2.4.6-p7 the mailing queue stopped working after a customer placed an order by including a new line character (CRLF) in their name at the checkout process.

### Steps to reproduce

1. Add an item to the shopping cart.
2. Click the Checkout button.
3. Fill in the customer address by providing a name including a new line character (CRLF).
4. Place the order.

### Expected result

The order is successfully placed, and Magento sends a message confirming the order. When the customer pays, Magento sends a message with the invoice, and when the item is shipped, Magento sends a shipping notification along with the tracking number.

### Actual result

The order is successfully placed, and Magento sends a message confirming the order. However, no message with the invoice is sent, after payment. In addition no shipping notification is sent after the order is shipped. Moreover, every time the Cron is executed, the below lines fill in the file **var/log/exception.log**:

`[2024-11-29T17:26:03.231902+00:00] main.WARNING: Could not add an invalid email address to the mailing queue {"exception":"[object] (Laminas\\Mail\\Exception\\InvalidArgumentException(code: 0): CRLF injection detected at /vendor/laminas/laminas-mail/src/Address.php:92)"} []`

### Additional information

The below error line appearing at the file **var/log/exception.log** is misleading as it gives the impression that the customer provided an invalid email address:

`[2024-11-29T17:26:03.231902+00:00] main.WARNING: Could not add an invalid email address to the mailing queue {"exception":"[object] (Laminas\\Mail\\Exception\\InvalidArgumentException(code: 0): CRLF injection detected at /vendor/laminas/laminas-mail/src/Address.php:92)"} []`

However, the line 92 of the file **/vendor/laminas/laminas-mail/src/Address.php** actually checks CRLF injection in the name, and not email address. I had to eliminate the CRLF from the customer name by directly edit the relevant fields at the database tables “**sales_order**” and “**sales_order_address**”. After the correction and Cron execution, all the pending invoice and shipping message were sent.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @dandrikop,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.

<img width="1429" alt="Screenshot 2024-12-04 at 10 04 31 am" src="https://github.com/user-attachments/assets/a22feff7-27f4-4122-81a9-ca16b3fdc4a3">


We have given a new line character for the first name we are getting as invalid first name.Could you please let us know if we are missing anything and also provide screenshots or screencast.

Thanks.

**Comment by dandrikop** [comment]
Hello @engcom-Bravo,

To be honest I don't know how the customer managed to pass a new line character into the "name" filed at the checkout page. However, it was sure that somehow he succeeded in doing that since I found the relevant record in the database tables “sales_order” and “sales_order_address” and corrected it.

**Comment by dandrikop** [comment]
Hello @engcom-Bravo,

I found the source of the problem. The name of the customer that included a new line character (CRLF) came from PayPal. Namely, the customer checked out via PayPal; i.e. without filling in the recipient address at the checkout page of Magento. So, the recipient address came to Magento from PayPal.

This was an extremely rare case. I have been working with Magento and PayPal since 2011, and I have never seen that problem in the past.

**Comment by engcom-Bravo** [comment]
Hi @dandrikop,

Thanks for your update.

We have verified the issue in Latest 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.


<img width="1421" alt="Screenshot 2024-12-11 at 10 46 19 am" src="https://github.com/user-attachments/assets/442f1d66-6497-4742-9ddd-b0bd23657c8f">



We have checked out via Paypal without filling in the recipient address at the checkout page of magento still we are getting error for First Name is not valid.

Thanks.
